File: graphql-test/src/main/java/com/junyss/graphqltest/api/team/model/dto/request/TeamRequestDto.java
package com.junyss.graphqltest.api.team.model.dto.request;

import javax.validation.constraints.NotBlank;

import lombok.Data;

@Data
public class TeamRequestDto {
	@NotBlank private Long teamId;
	@NotBlank private String manager;
	@NotBlank private String office;
	@NotBlank private String extensionNumber;
	private String mascot;
	@NotBlank private String cleaningDuty;
	private String project;
}

File: type-script-nest-graph-ql-실습/src/app/team/model/dto/request/team-request.dto.ts
import { IsNotEmpty, IsString } from "class-validator";
import { Field } from "@nestjs/graphql";
import { TeamEntity } from "../../entities/team.entity";

export class TeamRequestDto {
  @IsNotEmpty()
  @IsString()
  @Field(() => String)
  manager: string;

  @IsNotEmpty()
  @IsString()
  @Field(() => String)
  office: string;

  @IsNotEmpty()
  @IsString()
  @Field(() => String)
  extensionNumber: string;

  @IsString()
  @Field(() => String)
  mascot: string;

  @IsNotEmpty()
  @IsString()
  @Field(() => String)
  cleaningDuty: string;

  @IsString()
  @Field(() => String)
  project: string;

  toEntity(teamRequestDto: TeamRequestDto): TeamEntity {
    const team = new TeamEntity();
    team.manager = teamRequestDto.manager;
    team.office = teamRequestDto.office;
    team.extensionNumber = teamRequestDto.extensionNumber;
    team.mascot = teamRequestDto.mascot;
    team.cleaningDuty = teamRequestDto.cleaningDuty;
    team.project = teamRequestDto.project;
    return team;
  }
}
